src/wallet/wallet.h

```diff
@@ -607,16 +607,18 @@ struct CoinSelectionParams
     size_t change_output_size = 0;
     size_t change_spend_size = 0;
     CFeeRate effective_fee = CFeeRate(0);
+    CFeeRate m_long_term_feerate = CFeeRate(0);
     size_t tx_noinputs_size = 0;
     //! Indicate that we are subtracting the fee from outputs
     bool m_subtract_fee_outputs = false;
     bool m_avoid_partial_spends = false;
 
-    CoinSelectionParams(bool use_bnb, size_t change_output_size, size_t change_spend_size, CFeeRate effective_fee, size_t tx_noinputs_size, bool avoid_partial) :
+    CoinSelectionParams(bool use_bnb, size_t change_output_size, size_t change_spend_size, CFeeRate effective_fee, CFeeRate long_term_feerate, size_t tx_noinputs_size, bool avoid_partial) :
```

Comment:
In commit "wallet: Move long term feerate setting to CreateTransaction" (d3191194a8ced53ae511551dce1321c0416d9e6d)

In this commit and next commit, usage of SelectCoins is now more error prone because caller is responsible for setting `m_long_term_feerate` and `m_discard_feerate` variables in some cases (in other cases they are apparently never used), and there's not documentation about when to set them. Also these changes add to a proliferation of CoinSelectionParams constructor arguments can get easily confused together.

A safer approach might be to make these `std::optional<CFeeRate>` members instead of `CFeeRate` members so the CoinSelectionParams constructor doesn't have to be involved with them at all. Then they can be directly set by name, and explicit errors will be triggered if they are ever accessed without being set.

Reply:
The constructor parameters are more to make the tests easier to implement.